kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h

```diff
@@ -1458,24 +1387,81 @@ class ResidualBasedEliminationBuilderAndSolver
         }
     }
 
-    //**************************************************************************
+    /**
+     * @brief This method assembles the LHS of the system (on free rows)
+     * @param rA The LHS to assemble
+     * @param rLHSContribution The local LHS contribution
+     * @param rEquationId The equation id
+     */
+    void AssembleLHSCompleteOnFreeRows(
+        TSystemMatrixType& rA,
+        LocalSystemMatrixType& rLHSContribution,
+        EquationIdVectorType& rEquationId
+        )
+    {
+        const SizeType local_size = rLHSContribution.size1();
+
+        for (IndexType i_local = 0; i_local < local_size; ++i_local) {
+            const IndexType i_global = rEquationId[i_local];
 
-    void AssembleLHS_CompleteOnFreeRows(
-        TSystemMatrixType& A,
-        LocalSystemMatrixType& LHS_Contribution,
-        Element::EquationIdVectorType& EquationId
-    )
+            if (i_global < BaseType::mEquationSystemSize) {
+                AssembleRowContributionCompleteOnFreeDofs(rA, rLHSContribution, i_global, i_local, rEquationId);
+            }
+            //note that computation of reactions is not performed here!
+        }
+    }
+
+    /**
+     * @brief This function is equivalent to the AssembleRowContribution of the block builder and solver (designed for the AssembleLHSCompleteOnFreeRows)
+     * @note The main difference respect the block builder and solver is the fact that the fixed DoFs are skipped
+     */
+    inline void AssembleRowContributionCompleteOnFreeDofs(

```

Comment:
@loumalouomega to me it seems like this is also refactored
The rest seems fine for me, but I don't feel comfortable approving due to this

Reply:
I don't understand, I must remove this method?

Reply:
I would do this change in a separate PR, makes it easier to review

Reply:
Ok, removed and now is in concordance with the master